tests/conftest.py

```diff
@@ -37,53 +39,31 @@
 
 
 @pytest.fixture(scope="session")
-def pep518_wheelhouse(tmp_path_factory: pytest.TempPathFactory) -> Path:
-    wheelhouse = tmp_path_factory.mktemp("wheelhouse")
-
-    subprocess.run(
-        [
-            sys.executable,
-            "-m",
-            "pip",
-            "wheel",
-            "--wheel-dir",
-            str(wheelhouse),
-            f"{BASE}",
-        ],
-        check=True,
-    )
-    packages = [
-        "build",
-        "cython",
-        "hatchling",
-        "pip",
-        "setuptools",
-        "virtualenv",
-        "wheel",
-    ]
+def pep518_wheelhouse(pytestconfig: pytest.Config) -> Path:
+    wheelhouse = pytestconfig.cache.mkdir("wheelhouse")
+
+    main_lock = FileLock(wheelhouse / "main.lock")
+    with main_lock:
```

Comment:
I think you need to gate this by presence of scikit-build-core wheel.

Also don't think we need to 2 different locks?

Reply:
It's not an editable wheel, so it needs to be rebuilt every time the tests run - here it's being rebuilt by every process, but that's hopefully pretty cheap. Probably cheaper if we use build isolation.

Reply:
> It's testing the current source of scikit-build-core, so it has to be fresh. Yes, I'd like to only build it on one worker, but it's not that easy unless we make sure we always clean it up after a run (regardless of if the run failed, cancelled, etc).

I did some local debugging hoping that `hatch vcs` could help with the dirty files, but it seems like that would not work since it doesn't check the actual contents of the files when it is dirty, only the current date and if there are any dirty files.

The only alternative I can think of is to explicitly check against the `worker_id` and only run it on `master`, but are we able to do that without hard-coding `pytest-xdist` dependency?